tests/test_writer.py

```diff
@@ -2,30 +2,49 @@
 
 from defusedcsv import csv
 
+from unittest import TestCase
 
-def test_writer():
-    f = StringIO()
-    spamwriter = csv.writer(f, delimiter=' ', quotechar='|', quoting=csv.QUOTE_MINIMAL)
-    spamwriter.writerow(['Spam'] * 5 + ['@SUM(1+1)*cmd|\' /C calc\'!A0 '])
-    spamwriter.writerows([['Spam', 'Lovely Spam', 'Wonderful Spam'], ['A', 'B', '=3+4']])
 
-    f.seek(0)
-    assert f.read() == "Spam Spam Spam Spam Spam |'@SUM(1+1)*cmd\\||' /C calc'!A0 |\r\n" \
-                       "Spam |Lovely Spam| |Wonderful Spam|\r\nA B '=3+4\r\n"
+class WriterTests(TestCase):
```

Comment:
Instead of moving them into boilerplate classes, you can also check for assertions like this:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
True, that's another way to assert. I thought name-spacing the tests via classes wouldn't be bad. What do you prefer?